---
title: "Exploratory_and_Limma_analysis"
author: "Yanchao"
date: '2019-03-25'
output: github_document
---
## Load the library
```{r,warning=FALSE}
library(RColorBrewer)
library(cluster)
library(pvclust)
library(xtable)
library(limma)
library(plyr)
library(lattice)
library(RCurl)
options(download.file.method = "curl")
library(GEOquery)
library(knitr)
library(pheatmap)
  library(stringr)
library(ggplot2)
library(dplyr)
library(reshape2)
library(tidyverse)
```

## Load the data 
```{r}
combine_norm<- readRDS("combine_norm.rds")
Meta_data<- readRDS("Meta_data.rds")
```


## Plotting gene expression

```{r}
# melt into a format ggplot can easily work with
combine_norm2<-as.data.frame(combine_norm) %>% rownames_to_column("gene")
meltedExpressionMatrix <- combine_norm2 %>% melt(id = "gene") 

head_variable<-head(unique(meltedExpressionMatrix$variable),8)
meltedExpressionMatrix %>% filter(variable==head_variable) %>% 
  ggplot(aes(x = variable, y = value)) +
  geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+theme_bw()
```

By looking at the first 8 samples, it shows that the distribution of the sample are similar. 
```{r}
meltedExpressionMatrix %>% filter(variable==head_variable) %>% 
  ggplot(aes(x = value, color = variable)) +
  geom_density() +
  theme_bw()
```

The same result could be shown using the density plot.
## PCA
```{r}
#Scaled data
#scaled_combine_norm = scale(t(combine_norm))
spr_Dat <- t(scale(t(combine_norm)))
```

```{r}
#PCA on the Scaled data
PrinComps = prcomp(spr_Dat, center = FALSE, scale = FALSE)

plot(PrinComps)
```
```{r}
prop<-PrinComps$sdev^2/sum(PrinComps$sdev^2)
barplot(prop[1:10], xlab = 'Principal Components', ylab = 'Proportion of Variance Explained', main = 'Principal Component Analysis')
```

```{r}
# Get the subset of PCs that capture the most variance in your predictors
summary(PrinComps)
```

```{r}
# append the rotations for the first 10 PCs to the metadata
Meta_data_PrinCs = cbind(Meta_data, PrinComps$x[Meta_data$sample_name,1:10])

Meta_data_PrinCs_2 = melt(Meta_data_PrinCs, id=c("organism", "sample_name", "diagnosis", "age","batch"))

```

```{r}
#Relationships between age and PCs
ggplot(data=Meta_data_PrinCs_2, aes(x=age, col=age))+
  facet_wrap(~variable)+
  geom_point(aes(y=value))+
  labs(x="age", y="PC Loading", col="age")+
  theme_bw()
```
```{r}
#Relationships between autism and PCs
ggplot(data=Meta_data_PrinCs_2, aes(x=diagnosis, col=diagnosis))+
  facet_wrap(~variable)+
  theme(axis.text.x = element_text(angle=20))+
  geom_point(aes(y=value))+
  labs(x="diagnosis", y="PC Loading", col="diagnosis")+
  theme_bw()
```

```{r}
#Relationships between batch and PCs
ggplot(data=Meta_data_PrinCs_2, aes(x=batch, col=batch))+
  facet_wrap(~variable)+
  theme(axis.text.x = element_text(angle=20))+
  geom_point(aes(y=value))+
  labs(x="batch", y="PC Loading", col="batch")+
  theme_bw()
```

```{r}
# top 2 PCs related to age
ggplot(data=Meta_data_PrinCs, aes(x=PC1, y=PC2))+
  geom_point(aes(col=age))+
  labs(x="PC1", y="PC2", col="age")+
  theme_bw()
```
```{r}
# top 2 PCs related to autism
ggplot(data=Meta_data_PrinCs, aes(x=PC1, y=PC2))+
  geom_point(aes(col=diagnosis))+
  labs(x="PC1", y="PC2", col="diagnosis")+
  theme_bw()
```
```{r}
# top 2 PCs related to batch
ggplot(data=Meta_data_PrinCs, aes(x=PC1, y=PC2))+
  geom_point(aes(col=batch))+
  labs(x="PC1", y="PC2", col="batch")+
  theme_bw()
```

## Hierarchical Clustering for sample
```{r}
data_to_plot = spr_Dat

# compute pairwise distances
pr.dis <- dist(t(data_to_plot), method = "euclidean")

```

```{r}
pr.hc.a <- hclust(pr.dis, method = "average")

# plot them
plot(pr.hc.a, labels = FALSE, main = "Average", xlab = "")
```
```{r}
# set pheatmap clustering parameters
clust_dist_col = "euclidean"  #‘'correlation'’ for Pearson correlation, ‘'euclidean'’, ‘'maximum'’, ‘'manhattan'’, ‘'canberra'’, ‘'binary'’ or ‘'minkowski'’
clust_method = "average"  #‘'ward.D'’, ‘'ward.D2'’,‘'single'’, ‘'complete'’, ‘'average'’ (= UPGMA), ‘'mcquitty'’ (= WPGMA), ‘'median'’ (= WPGMC) or ‘'centroid'’ (= UPGMC)
clust_scale = "none"  #'column', 'none', 'row'

## the annotation option uses the covariate object (prDes) we defined. It should
## have the same rownames, as the colnames in our data object (data_to_plot).

pheatmap(data_to_plot, cluster_rows = FALSE, scale = clust_scale, clustering_method = clust_method, 
    clustering_distance_cols = clust_dist_col, show_colnames =FALSE, show_rownames = FALSE, 
    main = "Clustering heatmap for combine data", annotation = Meta_data[,c("diagnosis", "age","batch")])
```

## Limma
```{r}
head(spr_Dat %>% as_tibble())
Meta_data_1 = Meta_data
Design = Meta_data_1[,c("diagnosis", "age","batch")]
#Design Matrix
Des_Mat = model.matrix(~ diagnosis+age+batch, Design)
ds_Fit <- lmFit(spr_Dat, Des_Mat)
```

```{r}
ds_EbFit <- eBayes(ds_Fit)
ds_Hits <- topTable(ds_EbFit, coef = grep("diagnosis", colnames(coef(ds_EbFit))), p.value = 0.1, 
    n = Inf)
num_BHhits <- nrow(ds_Hits)

top_Genes <- rownames(ds_Hits)

# Scaled data of topGenes
top_Dat <- spr_Dat[top_Genes, ]

```


```{r}

top_10_Genes <- head(top_Genes,10)
transformGeneExpressionMatrix <- function(expressionMatrix) {
  expressionMatrix <- expressionMatrix %>%
    as.data.frame() %>% 
    column_to_rownames("gene") %>%
    t() %>% as.data.frame() %>% 
    rownames_to_column("sample_id") %>% 
    melt(id = "sample_id") %>% 
    as_tibble() %>% 
    select(sample_id,
           gene = variable, 
           expression = value)
  return(expressionMatrix)
}
Meta_data3 <-Meta_data %>% rownames_to_column("sample_id")
topGenes_ExpressionData <- combine_norm2 %>% 
  filter(gene %in% top_10_Genes) %>%
  transformGeneExpressionMatrix() %>% 
  left_join(Meta_data3, id = "sample_id")

#Examine differences in beta values among top 10 CGIs
ggplot(data=topGenes_ExpressionData, aes(x=diagnosis, y=expression, col=age))+
  geom_point()+
  geom_jitter() +
  stat_summary(aes(y=expression,group=1), fun.y = mean, geom = "line", color = "black")+
  facet_wrap(~gene)+
  labs(x=" diagnosis", y="top 10 gene expression")+
  theme_bw()
```


We start by using clustering algorithms to cluster the top 40 genes that showed differential expression across the different diagnosis stage (BH adjusted p value < 0.1).
```{r}
pheatmap(top_Dat, cluster_rows = TRUE, scale = "none", clustering_method = "average", 
    clustering_distance_cols = "euclidean", clustering_distance_rows = "euclidean", 
    annotation = Meta_data_1[,c("diagnosis", "age","batch")], show_rownames = FALSE,show_colnames  = FALSE)
```

```{r}
geneC_dis <- dist(top_Dat, method = "euclidean")

geneC_hc_a <- hclust(geneC_dis, method = "average")

plot(geneC_hc_a, labels = FALSE, main = "Hierarchical with Average Linkage", xlab = "")
```

